libs/ui/src/lib/dropdown-menu/index.tsx

```diff
@@ -0,0 +1,200 @@
+'use client';
+
+import * as React from 'react';
+import * as DropdownMenuPrimitive from '@radix-ui/react-dropdown-menu';
+import { CheckIcon, ChevronSmallRightIcon, CircleIcon } from '@storybook/icons';
+
+import { cn } from '@utils';
+
+const DropdownMenu = DropdownMenuPrimitive.Root;
+
+const DropdownMenuTrigger = DropdownMenuPrimitive.Trigger;
+
+const DropdownMenuGroup = DropdownMenuPrimitive.Group;
+
+const DropdownMenuPortal = DropdownMenuPrimitive.Portal;
+
+const DropdownMenuSub = DropdownMenuPrimitive.Sub;
+
+const DropdownMenuRadioGroup = DropdownMenuPrimitive.RadioGroup;
+
+const DropdownMenuSubTrigger = React.forwardRef<
+  React.ElementRef<typeof DropdownMenuPrimitive.SubTrigger>,
+  React.ComponentPropsWithoutRef<typeof DropdownMenuPrimitive.SubTrigger> & {
+    inset?: boolean;
+  }
+>(({ className, inset, children, ...props }, ref) => (
```

Comment:
If you don't mind some abstraction, we could create a utility, `styledComponent` (open to a better name), to handle the repetitive types for us:

```ts
// This could be in /libs/ui/src/lib/utils.ts ?
export function styledComponent<
  T extends React.ComponentType,
  P = Record<string, never>
>(
  ...args: Parameters<
    typeof React.forwardRef<
      React.ElementRef<T>,
      React.ComponentPropsWithoutRef<T> & P
    >
  >
) {
  return React.forwardRef<
    React.ElementRef<T>,
    React.ComponentPropsWithoutRef<T> & P
  >(...args);
}
```

Then you could define a styled component like so:

```suggestion
import { styledComponent } from '@ui';

const DropdownMenuSubTrigger = styledComponent<
  typeof DropdownMenuPrimitive.SubTrigger,
  {
    inset?: boolean;
  }
>(({ className, inset, children, ...props }, ref) => (
```

I find it much easier to read this way.

Reply:
These types are generated from ShadCn so I would not touch them - https://ui.shadcn.com/docs/components/dropdown-menu

Reply:
That's the beauty of ShadCn, you just copy and paste and you have your code with Tailwind ready without having it to be a dependency on anything.